### ISSUE TITLE
Failing test with GCC 12 and OpenBLAS 0.3.20 (zgemv_n_4.c)

### ISSUE THREAD

**Issue by mkrack** [cp2k/cp2k#2117]
The test `SIRIUS/regtest-1/He-full-potential.inp` gives with GCC 12.1 and OpenBLAS 0.3.20 the following SIGSEGV error in the routine `zgemv_n_4.c:499`

```
[get_singular_components] smallest eigen-value of the singular components:   0.0013665705096840
[diag_full_potential_first_variation_davidson] step: 0, current subspace size: 61, maximum subspace size: 341
[diag_full_potential_first_variation_davidson] step: 1, current subspace size: 101, maximum subspace size: 341
[diag_full_potential_first_variation_davidson] step: 2, current subspace size: 141, maximum subspace size: 341
[diag_full_potential_first_variation_davidson] step: 3, current subspace size: 181, maximum subspace size: 341
[diag_full_potential_first_variation_davidson] step: 4, current subspace size: 221, maximum subspace size: 341

Program received signal SIGSEGV: Segmentation fault - invalid memory reference.

Backtrace for this error:
#0  0x2b3ad79243ff in ???
#1  0x80bc617 in zgemv_n
	at ../kernel/x86_64/zgemv_n_4.c:499
#2  0x7eda21d in zgemv_
	at /data/user/krack/rt/gnu-latest-mpich/cp2k/tools/toolchain/build/OpenBLAS-0.3.20/interface/zgemv.c:263
#3  0x81bc379 in zlatrd_
	at /data/user/krack/rt/gnu-latest-mpich/cp2k/tools/toolchain/build/OpenBLAS-0.3.20/lapack-netlib/SRC/zlatrd.f:259
#4  0x81a67e2 in zhetrd_
	at /data/user/krack/rt/gnu-latest-mpich/cp2k/tools/toolchain/build/OpenBLAS-0.3.20/lapack-netlib/SRC/zhetrd.f:318
#5  0x81a3ddd in zheevx_
	at /data/user/krack/rt/gnu-latest-mpich/cp2k/tools/toolchain/build/OpenBLAS-0.3.20/lapack-netlib/SRC/zheevx.f:449
#6  0x75f6627 in _ZN18Eigensolver_lapack6solve_ISt7complexIdEEEiiiRN4sddk7dmatrixIT_EEPN4RealIS5_E4typeES7_
	at /data/user/krack/rt/gnu-latest-mpich/cp2k/tools/toolchain/build/SIRIUS-7.3.1/src/linalg/eigenproblem.hpp:208
#7  0x75f6a35 in _ZN18Eigensolver_lapack5solveEiiRN4sddk7dmatrixISt7complexIdEEEPdS5_
	at /data/user/krack/rt/gnu-latest-mpich/cp2k/tools/toolchain/build/SIRIUS-7.3.1/src/linalg/eigenproblem.hpp:135
#8  0x73baf0e in _ZNK6sirius4Band44diag_full_potential_first_variation_davidsonERNS_13Hamiltonian_kIdEE
	at /data/user/krack/rt/gnu-latest-mpich/cp2k/tools/toolchain/build/SIRIUS-7.3.1/src/band/diag_full_potential.cpp:657
#9  0x7392114 in _ZNK6sirius4Band20solve_full_potentialIdEEvRNS_13Hamiltonian_kIT_EE
	at /data/user/krack/rt/gnu-latest-mpich/cp2k/tools/toolchain/build/SIRIUS-7.3.1/src/band/solve.cpp:41
#10  0x73a04f7 in _ZNK6sirius4Band5solveIddEEvRNS_11K_point_setERNS_12Hamiltonian0IT_EEbd
	at /data/user/krack/rt/gnu-latest-mpich/cp2k/tools/toolchain/build/SIRIUS-7.3.1/src/band/solve.cpp:160
#11  0x741ebf5 in _ZN6sirius16DFT_ground_state4findB5cxx11Edddib
	at /data/user/krack/rt/gnu-latest-mpich/cp2k/tools/toolchain/build/SIRIUS-7.3.1/src/dft/dft_ground_state.cpp:225
#12  0x72fb5bd in operator()
	at /data/user/krack/rt/gnu-latest-mpich/cp2k/tools/toolchain/build/SIRIUS-7.3.1/src/api/sirius_api.cpp:1726
#13  0x72fb5bd in call_sirius<sirius_find_ground_state(void* const*, double const*, double const*, int const*, bool const*, bool*, int*, int*)::<lambda()> >
	at /data/user/krack/rt/gnu-latest-mpich/cp2k/tools/toolchain/build/SIRIUS-7.3.1/src/api/sirius_api.cpp:252
#14  0x72fb5bd in sirius_find_ground_state
	at /data/user/krack/rt/gnu-latest-mpich/cp2k/tools/toolchain/build/SIRIUS-7.3.1/src/api/sirius_api.cpp:1699
#15  0x72dbaca in __sirius_MOD_sirius_find_ground_state
	at /data/user/krack/rt/gnu-latest-mpich/cp2k/tools/toolchain/build/SIRIUS-7.3.1/src/api/sirius.f90:1755
#16  0xfaed45 in __sirius_interface_MOD_cp_sirius_energy_force
	at /data/user/krack/rt/gnu-latest-mpich/cp2k/src/sirius_interface.F:631
#17  0x20a3918 in __pwdft_environment_MOD_pwdft_calc_energy_force
	at /data/user/krack/rt/gnu-latest-mpich/cp2k/src/pwdft_environment.F:243
#18  0x16a4981 in __force_env_methods_MOD_force_env_calc_energy_force
	at /data/user/krack/rt/gnu-latest-mpich/cp2k/src/force_env_methods.F:276
#19  0x90250a in cp2k_run
	at /data/user/krack/rt/gnu-latest-mpich/cp2k/src/start/cp2k_runs.F:356
#20  0x903cbc in __cp2k_runs_MOD_run_input
	at /data/user/krack/rt/gnu-latest-mpich/cp2k/src/start/cp2k_runs.F:991
#21  0x8fd91d in cp2k
	at /data/user/krack/rt/gnu-latest-mpich/cp2k/src/start/cp2k.F:387
#22  0x559098 in main
	at /data/user/krack/rt/gnu-latest-mpich/cp2k/src/start/cp2k.F:44
```
This is possibly an issue of GCC 12.1.0 (compiler bug?), because that error does not appear with GCC 11.2.0.

**Comment by mkrack** [comment]
A similar error still occurs with OpenBLAS 0.3.21 and GCC 12.2.0 (and MPICH 3.4.3) using OpenMP (2 MPI ranks and 2 threads per rank). It might be related to this [OpenBLAS issue](https://github.com/xianyi/OpenBLAS/issues/3740).

```
Program received signal SIGSEGV: Segmentation fault - invalid memory reference.

Backtrace for this error:
#0  0x2b9aba7773ff in ???
#1  0xc2df5c6 in zgemv_n
	at ../kernel/x86_64/zgemv_n_4.c:499
#2  0xc09ab3c in zgemv_
	at /data/user/krack/rt/gnu-latest-mpich/cp2k/tools/toolchain/build/OpenBLAS-0.3.21/interface/zgemv.c:263
#3  0xbf3e484 in pzgemv_
	at /data/user/krack/rt/gnu-latest-mpich/cp2k/tools/toolchain/build/scalapack-2.1.0/PBLAS/SRC/pzgemv_.c:415
#4  0xbf16642 in pzlatrd_
	at /data/user/krack/rt/gnu-latest-mpich/cp2k/tools/toolchain/build/scalapack-2.1.0/SRC/pzlatrd.f:293
#5  0xbf08921 in pzhetrd_
	at /data/user/krack/rt/gnu-latest-mpich/cp2k/tools/toolchain/build/scalapack-2.1.0/SRC/pzhetrd.f:351
#6  0xbefe0cc in pzheevd_
	at /data/user/krack/rt/gnu-latest-mpich/cp2k/tools/toolchain/build/scalapack-2.1.0/SRC/pzheevd.f:355
#7  0x7571ba4 in __cp_cfm_diag_MOD_cp_cfm_heevd
	at /data/user/krack/rt/gnu-latest-mpich/cp2k/src/fm/cp_cfm_diag.F:122
#8  0x75747c8 in __cp_cfm_diag_MOD_cp_cfm_geeig
	at /data/user/krack/rt/gnu-latest-mpich/cp2k/src/fm/cp_cfm_diag.F:173
#9  0x35a2c73 in __qs_scf_diagonalization_MOD_do_general_diag_kp
	at /data/user/krack/rt/gnu-latest-mpich/cp2k/src/qs_scf_diagonalization.F:530
#10  0x4104344 in __qs_scf_loop_utils_MOD_qs_scf_new_mos_kp
	at /data/user/krack/rt/gnu-latest-mpich/cp2k/src/qs_scf_loop_utils.F:339
#11  0x3a60ab4 in __qs_scf_MOD_scf_env_do_scf
	at /data/user/krack/rt/gnu-latest-mpich/cp2k/src/qs_scf.F:491
#12  0x3a82283 in __qs_scf_MOD_scf
	at /data/user/krack/rt/gnu-latest-mpich/cp2k/src/qs_scf.F:243
#13  0x3020602 in __qs_energy_MOD_qs_energies
	at /data/user/krack/rt/gnu-latest-mpich/cp2k/src/qs_energy.F:93
#14  0x3424040 in qs_forces
	at /data/user/krack/rt/gnu-latest-mpich/cp2k/src/qs_force.F:202
#15  0x342c6fe in __qs_force_MOD_qs_calc_energy_force
	at /data/user/krack/rt/gnu-latest-mpich/cp2k/src/qs_force.F:112
#16  0x2bf6668 in __force_env_methods_MOD_force_env_calc_energy_force
	at /data/user/krack/rt/gnu-latest-mpich/cp2k/src/force_env_methods.F:271
#17  0x1645065 in __cp2k_debug_MOD_cp2k_debug_energy_and_forces
	at /data/user/krack/rt/gnu-latest-mpich/cp2k/src/cp2k_debug.F:170
#18  0x92e2fd in cp2k_run
	at /data/user/krack/rt/gnu-latest-mpich/cp2k/src/start/cp2k_runs.F:403
#19  0x938b32 in __cp2k_runs_MOD_run_input
	at /data/user/krack/rt/gnu-latest-mpich/cp2k/src/start/cp2k_runs.F:993
#20  0x928dbb in cp2k
	at /data/user/krack/rt/gnu-latest-mpich/cp2k/src/start/cp2k.F:379
#21  0x566848 in main
	at /data/user/krack/rt/gnu-latest-mpich/cp2k/src/start/cp2k.F:44
BLACS ERROR 'MPI error 944353381 on call to MPI_Isend'
from {1,0}, pnum=1, Contxt=1, on line 16 of file '/data/user/krack/rt/gnu-latest-mpich/cp2k/tools/toolchain/build/scalapack-2.1.0/BLACS/SRC/BI_Asend.c'.

application called MPI_Abort(MPI_COMM_WORLD, 1) - process 1
srun: Job step aborted: Waiting up to 32 seconds for job step to finish.
srun: error: merlin-c-023: task 0: Segmentation fault
srun: error: merlin-c-024: task 1: Killed


Runtime failure with code 139.
```

**Comment by oschuett** [comment]
This was presumably fixed by the upgrade to OpenBLAS 0.3.21 via #2236.
